src/silverlabnwb/metadata.py

```diff
@@ -35,39 +36,69 @@ def read_user_config():
     defaults.
     """
     with pkg_resources.resource_stream(__name__, 'metadata.yaml') as defaults:
-        settings = read_config_file(defaults)
+        settings, comments = read_base_config_file(defaults)
     if os.path.isfile(user_conf_path):
         with open(user_conf_path, 'r') as user_params:
-            settings = read_config_file(user_params, settings)
-    return settings
+            settings = update_config_file(user_params, settings)
+    return settings, comments
+
+
+def read_base_config_file(stream):
+    yaml_reader = yaml.YAML()
+    parsed_yaml_data = yaml_reader.load(stream)
+    yaml_data_comments = read_comments(parsed_yaml_data)
+    settings = {}
+    recursive_dict_update(settings, strip_strings(parsed_yaml_data))
+    return settings, yaml_data_comments
+
+
+def read_comments(settings):
+    comments = {}
+    for k, v in settings.items():
+        if isinstance(v, collections.Mapping):
+            comments[k] = read_comments(v)
+        elif isinstance(v, list):
+            comments[k] = [read_comments(entry) for entry in v]
+        else:
+            comment = settings.ca.items.get(k, None)
+            if comment is not None:
+                yaml_comment = comment[2].value  # access CommentToken
+
+                def should_keep(line):
+                    return len(line) > 0 and not line.startswith('>')
```

Comment:
We could perhaps move this further up (say, just before/after declaring `comments`) so we don't redefine the function for every item.